### PR TITLE
replaced namedlist and ordereddefaultdict with stdlib implementations

### PR DESCRIPTION
**Pull request by HeliosLHC**
This is part of series of PRs I'm planning on making to split apart the numerous changes I have made in my fork (support for Python 3.10+ along with other modernizations) into smaller pieces.

This PR accounts for changes to the Python Standard Library to:
1. Replace namedlist packaged with stdlib collections.namedtuple
2. Replaced custom ordereddefaultdict class with stdlib coillections.defaultdict (dicts are now officially ordered)
3. Replace references to use stdlib collections.abc (many classes were moved to this new module)

@ivan  You can find me as Terbium on HackInt as it's been a while since we chatted.

### CODE REVIEW

**Comment by Arkiver2** [comment]
Should these updates be made to https://github.com/ArchiveTeam/wpull as well? @JustAnotherArchivist

**Comment by HeliosLHC** [comment]
> Should these updates be made to https://github.com/ArchiveTeam/wpull as well? @JustAnotherArchivist

My fork has quite a massive list of modernizations and drops support for a lot of Python versions (requires 3.10+) which Ivan was ok with. I believe JAA indicated that due to my fork based ludios_wpull based.

@Arkiver2  I spoke with JAA on this 6 or so months ago and it was indicated that as it was from Ivan's fork it won't be upstreamed which is fine with me as this was mostly a private fork I maintained.

File: wpull/warc/recorder.py [hunk 1]
```diff
@@ -31,21 +31,10 @@
...
-        ('appending', False),
-        ('digests', True),
-        ('cdx', None),
-        ('max_size', None),
-        ('move_to', None),
-        ('url_table', None),
-        ('software_string', None)
-    ]
+    ('compress','extra_fields','temp_dir','log','appending','digests','cdx','max_size','move_to','url_table','software_string'),
+    defaults=(True, None, './', True, False,True, None,None,None,None,None)
```
Comment (ivan) [inline wpull/warc/recorder.py:37]: I don't like how the defaults are specified for a namedtuple, especially the wide one here. Is there a better way to do this with [attrs](https://www.attrs.org/en/stable/) or [dataclasses](https://docs.python.org/3/library/dataclasses.html) or similar?

**Comment by HeliosLHC** [comment]
Replaced additional usages of namedtuples with dataclasses, and created new python-3.11 branch to merge these changes into as there are quite a lot of other changes I need to merge in from my private repo to support compatibility with Python 3.11

**Pull request merged**